src/modules/ekf2/EKF/ekf.cpp

```diff
@@ -250,6 +250,29 @@ void Ekf::predictState()
 	_state.quat_nominal = (_state.quat_nominal * dq).normalized();
 	_R_to_earth = Dcmf(_state.quat_nominal);
 
+	// if there's accel clipping (asymmetric railing) then for convenience register in earth frame
+	if (_imu_sample_delayed.delta_vel_clipping[0] || _imu_sample_delayed.delta_vel_clipping[1] || _imu_sample_delayed.delta_vel_clipping[2]) {
+
+		// delta velocity clipping in body fixed frame
+		const Vector3f clipping_bf {
+			_imu_sample_delayed.delta_vel_clipping[0] ? 1.f : 0.f,
+			_imu_sample_delayed.delta_vel_clipping[1] ? 1.f : 0.f,
+			_imu_sample_delayed.delta_vel_clipping[2] ? 1.f : 0.f
+		};
+
+		// delta velocity clipping in earth frame
+		const Vector3f clipping_ef = _R_to_earth * clipping_bf;
```

Comment:
With a minor change to this LOC, could you collapse this block into a for loop? (I don't know if that has a negative performance impact, you can speak to that.)  Nice work Daniel.